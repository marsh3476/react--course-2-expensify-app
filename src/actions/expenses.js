import { v4 as uuidv4 } from 'uuid';

// ADD EXPENSE
export const addExpense = (
  { 
    description = '', 
    note = '', 
    amount = 0, 
    createdAt=0 
  } = {}
) => ({
  type: 'ADD_EXPENSE',
  expense: {
    id: uuidv4(),
    description: description,
    note: note, 
    amount: amount,
    createdAt: createdAt
  }
});

// EDIT EXPENSE
export const editExpense = (id, updates) => ({
  type: 'EDIT_EXPENSE',
  id: id,
  updates: updates
});
// REMOVE EXPENSE
export const removeExpense = ({ id } = {}) => {
  return {
    type: 'REMOVE_EXPENSE',
    id: id
  }
  
};