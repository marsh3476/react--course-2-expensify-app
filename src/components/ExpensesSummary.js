import React from 'react';
import { connect } from 'react-redux';
import selectExpensesByFilters from '../selectors/expenses';
import getExpenseTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export class ExpensesSummary extends React.Component {

  render() {
    return (
      <h3>Viewing {this.props.filteredExpenses.length} expense(s) totalling {numeral(
                      this.props.getExpenseTotal(this.props.filteredExpenses) / 100)
                      .format('$0,0.00')}</h3>
    );
  }


}

const mapStateToProps = (state) => {
  return {
    filteredExpenses: selectExpensesByFilters(state.expenses, state.filters),
    getExpenseTotal: getExpenseTotal
  }
}
export default connect(mapStateToProps)(ExpensesSummary);
