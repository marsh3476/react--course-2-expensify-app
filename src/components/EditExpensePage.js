import { connect } from 'react-redux';
import { editExpense, removeExpense } from '../actions/expenses';
import React from 'react';
import ExpenseForm from './ExpenseForm';
 

export class EditExpensePage extends React.Component {

  onSubmit = (expense) => {
    this.props.editExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  };
  onClick = (e) => {
    this.props.removeExpense(this.props.expense)
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <ExpenseForm
          expense={this.props.expense} 
          onSubmit={this.onSubmit}
        />
        <button onClick={this.onClick}>Remove</button>
      </div>
    );
  }

}

const mapDispatchToProps = (dispatch) => {
  return {
    editExpense: (id, expense) => dispatch(editExpense(id, expense)),
    removeExpense: (expense) => dispatch(removeExpense(expense))
  };
};

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => 
      expense.id === props.match.params.id
    )
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);