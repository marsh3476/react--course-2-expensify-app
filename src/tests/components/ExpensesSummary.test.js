import { ExpensesSummary } from '../../components/ExpensesSummary';
import { shallow } from 'enzyme';
import React from 'react';
import expenses from '../fixtures/expenses';


test('should render Expenses summary for single expense', () => {
  const wrapper = shallow(<ExpensesSummary 
                          filteredExpenses={[expenses[0]]}
                          getExpenseTotal={() => 1500} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render Expenses summary for multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary 
                          filteredExpenses={expenses}
                          getExpenseTotal={() => 1500} />);
  expect(wrapper).toMatchSnapshot();
});