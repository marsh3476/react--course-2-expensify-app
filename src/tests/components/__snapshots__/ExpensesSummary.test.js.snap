// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expenses summary for multiple expenses 1`] = `
<h1>
  Viewing 
  3
   expense(s) totalling 
  $15.00
</h1>
`;

exports[`should render Expenses summary for single expense 1`] = `
<h1>
  Viewing 
  1
   expense(s) totalling 
  $15.00
</h1>
`;
